src/service.rs

```diff
@@ -1,123 +0,0 @@
-use tower_service::Service;
-
-use http::{Request, Response};
-use futures::{Future, Poll};
-
-use self::sealed::Sealed;
-
-/// An HTTP service
-///
-/// This is not intended to be implemented directly. Instead, it is a trait
-/// alias of sorts. Implements the `tower_service::Service` trait using
-/// `http::Request` and `http::Response` types.
-pub trait HttpService<RequestBody>: Sealed<RequestBody> {
-    /// Response payload.
-    type ResponseBody;
-
-    /// Errors produced by the service.
-    type Error;
-
-    /// The future response value.
-    type Future: Future<Item = Response<Self::ResponseBody>, Error = Self::Error>;
-
-    /// Returns `Ready` when the service is able to process requests.
-    fn poll_ready(&mut self) -> Poll<(), Self::Error>;
-
-    /// Process the request and return the response asynchronously.
-    fn call(&mut self, request: Request<RequestBody>) -> Self::Future;
-
-    /// Wrap the HttpService so that it implements tower_service::Service
-    /// directly.
-    ///
-    /// Since `HttpService` does not directly implement `Service`, if an
-    /// `HttpService` instance needs to be used where a `T: Service` is
-    /// required, it must be wrapped with a type that provides that
-    /// implementation. `LiftService` does this.
-    fn lift(self) -> LiftService<Self> where Self: Sized {
-        LiftService { inner: self }
-    }
-
-    /// Same as `lift` but operates on an HttpService reference.
-    fn lift_ref(&mut self) -> LiftServiceRef<Self> where Self: Sized {
-        LiftServiceRef { inner: self }
-    }
-}
-
-/// Wraps an `HttpService` instance, implementing `tower_service::Service`.
-///
-/// See [`lift`] function documentation for more details.
-///
-/// [`lift`]: #
-pub struct LiftService<T> {
```

Comment:
Some downstream services like @LucioFranco 's tower-consul breaks with lift service being removed, lift should just be replaced with straight service impls correct?

Reply:
lift just got renamed to `AsService` so just need to change the functions we call in tower-consul

Reply:
There is more to it than that, LiftService only has T, where AsService is `AsService<'a, T>`

Reply:
* `LiftService` -> `IntoService`
* `LiftServiceRef` -> `AsService`.

It should have just been a rename. If not, I messed up.